webapp/components/survey/NodeDefDetails/store/actions/useSaveNodeDefEdits.js

```diff
@@ -0,0 +1,68 @@
+import * as Survey from '@core/survey/survey'
+import * as NodeDef from '@core/survey/nodeDef'
+import * as NodeDefLayout from '@core/survey/nodeDefLayout'
+import * as SurveyValidator from '@core/survey/surveyValidator'
+
+import { LoaderActions, NotificationActions } from '@webapp/store/ui'
+
+import * as SurveyState from '@webapp/store/survey/state'
+import * as NodeDefsActions from '@webapp/store/survey/nodeDefs/actions'
+
+import * as NodeDefState from '../state'
+
+// Persists the temporary changes applied to the node def in the state
+export const useSaveNodeDefEdits = ({ nodeDefState, setNodeDefState }) => () => async (dispatch, getState) => {
```

Comment:
this should not return a thunk.
Do not mix local actions with redux actions

Reply:
ok